cmd/search/httpchartpng.go

```diff
@@ -75,6 +75,10 @@ func (o *options) handleChartPNG(w http.ResponseWriter, req *http.Request) {
 	maxDuration := 0
 	scatters := make([]*scatter, len(index.Search)+3)
 	for _, job := range jobs {
+		if index.Job != nil && !index.Job.MatchString(job.Spec.Job) {
```

Comment:
This isn't set on indexed jobs.  We'll have to extract it inside of indexgcs.go.

Reply:
`Spec.Job` isn't set?  Then when the user requests an `index.Job` filter, we want to exclude those `Spec.Job`-less jobs from the PNG, right?

Reply:
Right now the other code are doing that by looking at the file on disk.

All the backfilled responses don't have anything in spec set.  You can infer it from the URL, which is fine (but ugly, the code should be split out enough to be the same in multiple places), and then set it there.